[auth-session] Extract AuthSession into unimodule (#6989)
expo/expo: docs/pages/versions/unversioned/sdk/auth-session.md

Get the URL that your authentication provider needs to redirect to. For example: `https://auth.expo.io/@your-username/your-app-slug`.

> **Note** This method will throw an exception if you're using the bare workflow.

## Usage in the bare React Native app

> In managed apps, `AuthSession` uses expo servers to create a proxy between your application and the auth provider. Unfortunately, you don't have access to these servers in bare apps. To overcome this problem, you can create your proxy service.

Reviewer: It's better to say that we don't provide support for something rather than that you don't have access to something - the later has more negative meaning.
```suggestion
In managed apps, `AuthSession` uses Expo servers to create a proxy between your application and the auth provider. Unfortunately, we don't provide support to use these servers in bare apps. To overcome this, you can create your proxy service.
```

### Proxy Service

This service is responsible for:

- redirecting traffic from your application to the authentication service
- redirecting response from the auth service to your application using a deep link

To better understand how it works, check out this implementation in `node.js`:

```js
const http = require('http');
const url = require('url');

const PORT = PORT;
const DEEP_LINK = DEEP_LINK_TO_YOUR_APPLICATION;

function redirect(response, url) {
  response.writeHead(302, {
    Location: url,
  });
  response.end();
}

http
  .createServer((request, response) => {
    // get parameters from request
    const parameters = url.parse(request.url, true).query;

    // if parameters contain authServiceUrl, this request comes from the application
    if (parameters.authServiceUrl) {
      // redirect user to the authUrl
      redirect(response, decodeURIComponent(parameters.authServiceUrl));
      return;
    }

    // redirect response from the auth service to your application
    redirect(response, DEEP_LINK);
  })
  .listen(PORT);
```

> Client code which work with this service:

Reviewer: ```suggestion
Client code which works with this service:
```
